core/src/main/java/hudson/model/Job.java

```diff
@@ -1210,7 +1210,8 @@ public synchronized void doConfigSubmit(StaplerRequest req,
 
             String newName = req.getParameter("name");
             final ProjectNamingStrategy namingStrategy = Jenkins.getInstance().getProjectNamingStrategy();
-            if (newName != null && !newName.equals(name)) {
+            if (newName != null && !newName.trim().equals(name.trim())) {
```

Comment:
Thinks it can be writed with help of StringUtils from commons-lang

with static import it can be:
```java
import static org.apache.commons.lang.StringUtils.trim;

//...
if(!StringUtils.equals(trim(name), trim(newName))) {
    newName = trim(newName);
}
```